ai-platform-unified/notebooks/official/featurestore_movie_prediction.ipynb

```diff
@@ -0,0 +1,1277 @@
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "qwznuUiwjwJF"
+      },
+      "source": [
+        "# Start to import, will take a couple of minutes\n",
+        "ingestion_lro = admin_client.import_feature_values(import_users_request)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "_sDl3ZcrF64T"
+      },
+      "source": [
+        "# Polls for the LRO status and prints when the LRO has completed\n",
+        "ingestion_lro.result()"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "laXdJPIqkLJO"
+      },
+      "source": [
+        "### Import feature values for Movies\n",
+        "\n",
+        "Similarly, import feature values for 'movies' to the feature store.\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "-W1lCxgDl6iR"
+      },
+      "source": [
+        "import_movie_request = featurestore_service_pb2.ImportFeatureValuesRequest(\n",
+        "    entity_type=admin_client.entity_type_path(PROJECT_ID, LOCATION,\n",
+        "                                              FEATURESTORE_ID, \"movies\"),\n",
+        "    avro_source=io_pb2.AvroSource(\n",
+        "        gcs_source=io_pb2.GcsSource(uris=[\"gs://cloud-samples-data-us-central1/ai-platform-unified/datasets/featurestore/movies.avro\"])\n",
+        "    ),\n",
+        "    entity_id_field=\"movie_id\",\n",
+        "    feature_specs=[\n",
+        "        featurestore_service_pb2.ImportFeatureValuesRequest.FeatureSpec(\n",
+        "            id=\"title\"),\n",
+        "        featurestore_service_pb2.ImportFeatureValuesRequest.FeatureSpec(\n",
+        "            id=\"genres\"),\n",
+        "        featurestore_service_pb2.ImportFeatureValuesRequest.FeatureSpec(\n",
+        "            id=\"average_rating\"),\n",
+        "    ],\n",
+        "    feature_time_field=\"update_time\",\n",
+        "    worker_count=10)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Q-MATtpvm3HI"
+      },
+      "source": [
+        "# Start to import, will take a couple of minutes\n",
+        "ingestion_lro = admin_client.import_feature_values(import_movie_request)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "VpaK3yRCnNnI"
+      },
+      "source": [
+        "# Polls for the LRO status and prints when the LRO has completed\n",
+        "ingestion_lro.result()"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "9TdxPYdDXjnA"
+      },
+      "source": [
+        "## Online serving"
```

Comment:
a high-level comment: if we add an example of actually training a model, as discussed, then I think the flow would work better to first show the batch serving case, then train the model using the batch data, then the online serving API case, showing how to construct a prediction request. 
[For simplicity I don't think the model would have to be saved and deployed; it could just be an in-notebook object]

Reply:
IIRC, The reason why we put online serving ahead of batch, in this standalone notebook, is that online serving is the most-asked feature from customer.

But I agree that a data-ingestion + model training + model serving version tells a wonderful e2e story. The only question is if we consider the tight timeline before I/O, I think most likely we will use this as a standalone notebook.